jobs/ftp-poller/tasks/poll_ftp_task.py

```diff
@@ -0,0 +1,133 @@
+
+            file_list: List[SFTPAttributes] = sftp_client.listdir_attr(ftp_dir)
+            current_app.logger.info(f'Found {len(file_list)} to be copied.')
+            for file in file_list:
+                file_name = file.filename
+                file_full_name = ftp_dir + '/' + file_name
+                current_app.logger.info(f'Processing file  {file_full_name} started-----.')
+                if PollFtpTask._is_valid_payment_file(file_full_name):
+                    cls._upload_to_minio(file, file_full_name, sftp_client)
+                    payment_file_list.append(file_name)
+
+            if len(payment_file_list) > 0:
+                PollFtpTask._post_process(payment_file_list)
+
+        except Exception as e:  # pylint: disable=broad-except
+            current_app.logger.error(e)
+        finally:
+            SFTPService.get_connection().close()
+        return payment_file_list
+
+    @classmethod
+    def _upload_to_minio(cls, file, file_full_name, sftp_client):
+        """Upload to minio."""
+        f: SFTPFile
+        with sftp_client.open(file_full_name) as f:
+            f.prefetch()
+            value_as_bytes = f.read()
+            try:
+                MinioService.put_object(value_as_bytes, file.filename, file.st_size)
+            except Exception as e:  # pylint: disable=broad-except
+                current_app.logger.error(e)
+                current_app.logger.error(f'upload to minio failed for the file: {file_full_name}')
+                raise
+
+    @classmethod
+    def _post_process(cls, payment_file_list: List[str]):
+        """
+        Post processing of the file.
+
+        1.Move the file to backup folder
+        2.Send a message to queue
+        """
+        cls._move_file_to_backup(payment_file_list)
+        cls._publish_to_queue(payment_file_list)
+
+    @classmethod
+    def _move_file_to_backup(cls, payment_file_list):
+        sftp_client = SFTPService.get_connection()
+        ftp_backup_dir: str = current_app.config.get('CAS_SFTP_BACKUP_DIRECTORY')
+        ftp_dir: str = current_app.config.get('CAS_SFTP_DIRECTORY')
+        for file_name in payment_file_list:
+            sftp_client.rename(ftp_dir + '/' + file_name, ftp_backup_dir + '/' + file_name)
+
+    @classmethod
+    def _is_valid_payment_file(cls, file_name):
+        sftp_client = SFTPService.get_connection()
+        return sftp_client.isfile(file_name)
+
+    @classmethod
+    def _publish_to_queue(cls, file_names_list: List[str]):
+        # Publish message to the Queue, saying file has been uploaded. Using the event spec.
+        file_names: str = ','.join(file_names_list)
+        queue_data = {
+            'fileName': file_names,
+            'file_source': 'MINIO',
+            'location': current_app.config['MINIO_BUCKET_NAME']
+        }
+
+        payload = {
+            'specversion': '1.x-wip',
+            'type': 'bc.registry.payment.' + 'paymentFileTypeUploaded',
```

Comment:
Also can this be `bc.registry.payment.paymentFileUploaded`, there is a Type in the string. Not sure about its significance.

Reply:
done